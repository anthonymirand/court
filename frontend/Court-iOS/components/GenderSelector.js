import React from 'react';
import { Icon } from 'expo';

import { Platform, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Colors from '../constants/Colors';

export default class GenderSelector extends React.Component {
  constructor(props) {
    super(props);
    this.state = { selfSelected: null, otherSelected: null };
  }

  genderSelected(gender) {
    this.setState({ selfSelected: gender });
    this.props.onGenderSelected(gender);
  }

  preferenceSelected(gender) {
    this.setState({ otherSelected: gender });
    this.props.onPreferenceSelected(gender);
  }

  render() {
    return (
      // Select your gender
      <View style={styles.outerView}>
        <Text style={styles.promptText}>{"I am:"}</Text>
        // Selector
        <View style={styles.selector}>
          <TouchableOpacity onPress={() => this.genderSelected('male')}>
            <Text style={this.state.selfSelected == 'male' ? styles.selected : styles.unselected}>
              Male
            </Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.genderSelected('female')}>
            <Text style={this.state.selfSelected == 'female' ? styles.selected : styles.unselected}>
              Female
            </Text>
          </TouchableOpacity>
        </View>

        {this.state.selfSelected && (
          <View style={{ alignItems: 'center', justifyContent: 'center' }}>
            // Select preferred gender
            <Text style={styles.promptText}>{"Looking for:"}</Text>
            // Selector
            <View style={styles.selector}>
              <TouchableOpacity onPress={() => this.preferenceSelected('male')}>
                <Text style={this.state.otherSelected == 'male' ? styles.selected : styles.unselected}>
                  Males
                </Text>
              </TouchableOpacity>
              <TouchableOpacity onPress={() => this.preferenceSelected('female')}>
                <Text style={this.state.otherSelected == 'female' ? styles.selected : styles.unselected}>
                  Females
                </Text>
              </TouchableOpacity>
              <TouchableOpacity onPress={() => this.preferenceSelected('both')}>
                <Text style={this.state.otherSelected == 'both' ? styles.selected : styles.unselected}>
                  Both
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  outerView: {
    marginTop: 5,
    marginBottom: 45,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  promptText: {
    marginTop: 40,
    fontSize: 30,
    color: 'white',
    marginBottom: 15,
  },
  selector: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  selected: {
    marginLeft: 15,
    fontSize: 30,
    color: Colors.teal,
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: 'white',
    borderRadius: 10,
    overflow: 'hidden',
    padding: 5,
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { height: 8 },
        shadowOpacity: 0.15,
        shadowRadius: 9,
      },
      android: {
        elevation: 20,
      },
    }),
  },
  unselected: {
    borderWidth: 1,
    borderColor: 'rgba(255, 255, 255, 0.7)',
    borderRadius: 10,
    overflow: 'hidden',
    padding: 5,
    marginLeft: 15,
    fontSize: 20,
    color: 'rgba(255, 255, 255, 0.7)',
  },
});
