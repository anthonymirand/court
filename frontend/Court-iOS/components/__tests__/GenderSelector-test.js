import 'react-native';
import React from 'react';
import GenderSelector from '../GenderSelector';
import renderer from 'react-test-renderer';

it('Renders GenderSelector', () => {
  const mocks = { onGenderSelected: jest.fn(), onPreferenceSelected: jest.fn() };

  const tree = renderer.create(
    <GenderSelector
      onGenderSelected={mocks.onGenderSelected}
      onPreferenceSelected={mocks.onPreferenceSelected}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
