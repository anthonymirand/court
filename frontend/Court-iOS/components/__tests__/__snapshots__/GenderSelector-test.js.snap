// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders GenderSelector 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 45,
      "marginTop": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "marginBottom": 15,
        "marginTop": 40,
      }
    }
  >
    I am:
  </Text>
  // Selector
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.7)",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 20,
            "marginLeft": 15,
            "overflow": "hidden",
            "padding": 5,
          }
        }
      >
        Male
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderColor": "rgba(255, 255, 255, 0.7)",
            "borderRadius": 10,
            "borderWidth": 1,
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 20,
            "marginLeft": 15,
            "overflow": "hidden",
            "padding": 5,
          }
        }
      >
        Female
      </Text>
    </View>
  </View>
</View>
`;
